osidb/serializer.py

```diff
@@ -352,7 +354,50 @@ def validate_acl(self, embargoed):
                 )
 
 
-class ACLMixinSerializer(serializers.ModelSerializer):
+class BaseSerializer(serializers.ModelSerializer):
+    """
+    base serializer class which should be inherited by every serializer
+    of a model which save method requires any additional parameters
+
+    the reason is that the Django ModelSerializer does not provide any way
+    how to pass these parameters through create or update methods and then
+    those need to be called multiple times repeating the same actions and
+    complicating the whole save machinery
+    """
+
+    # TODO rewrite create machinery to use save
+    # def create(self, validated_data, *args, **kwargs):
+
+    def update(self, instance, validated_data, *args, **kwargs):
+        """
+        exteded standard Django REST framework update method
+        optionally calling save with additional parameters
+        """
+        raise_errors_on_nested_writes("update", self, validated_data)
+        info = model_meta.get_field_info(instance)
+
+        m2m_fields = []
+        for attr, value in validated_data.items():
+            if attr in info.relations and info.relations[attr].to_many:
+                m2m_fields.append((attr, value))
+            else:
+                setattr(instance, attr, value)
+
+        # the additional arguments to the following save call are the only difference from the original
+        # https://github.com/encode/django-rest-framework/blob/master/rest_framework/serializers.py#L1018
```

Comment:
We should reference the exact version (permalink or probably even better tag) as master will change in the future and it will not be clear from exactly where this code derived.

Also by doing this we overwrite some core functionality code in DRF, if this for whatever reason changes in the furure with the new versions, we don't get the changes. I am thinking how can we ensure that we will remember to check this/update this.

Reply:
Very good point. I am referencing that line in the version 3.15.2 which is the one we use. I am not sure how to make sure we account for any significant change in the source here. I assume it should be pretty rare but we cannot know exactly. For now however it seems to me the right/easiest way how to push the necessary arguments to our custom `save` machinery avoiding the current workarounds in the serializer. I would say that the low risk of a future break is worth fixing the bad serializer-model integration we have.

Reply:
I agree. Lets focus on the serializer offloading in the future.